test/sdk/TopologyFixture.h

```diff
@@ -0,0 +1,76 @@
+/********************************************************************************
+ *    Copyright (C) 2019 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#ifndef FAIR_MQ_TEST_TOPOLOGYFIXTURE
+#define FAIR_MQ_TEST_TOPOLOGYFIXTURE
+
+#include <DDS/Tools.h>
+#include <DDS/Topology.h>
+#include <TestEnvironment.h>
+#include <fairlogger/Logger.h>
+#include <gtest/gtest.h>
+#include <string>
+
+namespace fair {
+namespace mq {
+namespace test {
+
+struct Topology : ::testing::Test
+{
+    Topology()
+        : mDDSTopologyFile(SDK_TESTSUITE_SOURCE_DIR "/test_topo.xml")
+        , mDDSSessionID()
+    {}
+
+    auto SetUp() -> void override {
+        mDDSSessionID = mDDSSession.create();
+
+        mDDSTopology = std::make_shared<dds::topology_api::CTopology>();
+        mDDSTopology->init(mDDSTopologyFile);
+        auto requiredNofAgents(mDDSTopology->getRequiredNofAgents());
+        std::string ddsRmsPlugin("localhost");
+
+        dds::tools_api::SSubmitRequestData submitInfo;
+        submitInfo.m_rms = ddsRmsPlugin;
+        submitInfo.m_instances = requiredNofAgents;
+        auto submitRequest = dds::tools_api::SSubmitRequest::makeRequest(submitInfo);
+        submitRequest->setMessageCallback(
+            [](const dds::tools_api::SMessageResponseData& _message) {
+            LOG(info) << "Server reports: " << _message.m_msg;
+        });
+        submitRequest->setDoneCallback([]() {
+            LOG(info) << "Agents submitted";
+        });
+
+        LOG(info) << "DDSSessionID:        " << mDDSSessionID;
+        LOG(info) << "DDSTopologyFile:     " << mDDSTopologyFile;
+        LOG(info) << "requiredNofAgents:   " << requiredNofAgents;
+        LOG(info) << "DDSRMSPlugin:        " << ddsRmsPlugin;
+        LOG(info) << "DDSSession running?: " << mDDSSession.IsRunning();
+
+        mDDSSession.sendRequest<dds::tools_api::SSubmitRequest>(submitRequest);
+        mDDSSession.blockCurrentThread();
+
+        // TODO activate topology
+    }
```

Comment:
@AndreyLebedev When I am running the `SetUp()` method above, the current thread seems to not be blocked and I never see the registered callbacks fired:

```
╭─dklein@gamma ~/projects/FairMQ/build ‹ruby-2.6.0› ‹sdk_topology›
╰─➤  ctest -R SDK.Topology -VV                                                                                                        130 ↵
UpdateCTestConfiguration  from :/home/dklein/projects/FairMQ/build/DartConfiguration.tcl
Parse Config file:/home/dklein/projects/FairMQ/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/dklein/projects/FairMQ/build/DartConfiguration.tcl
Parse Config file:/home/dklein/projects/FairMQ/build/DartConfiguration.tcl
Test project /home/dklein/projects/FairMQ/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 137
    Start 137: SDK.Topology.Basic

137: Test command: /home/dklein/projects/FairMQ/build/test/testsuite_SDK "--gtest_filter=Topology.Basic" "--gtest_also_run_disabled_tests"
137: Test timeout computed to be: 5
137: Note: Google Test filter = Topology.Basic
137: [==========] Running 1 test from 1 test case.
137: [----------] Global test environment set-up.
137: [----------] 1 test from Topology
137: [ RUN      ] Topology.Basic
137: [INFO] DDSSessionID:        e843ae11-a393-440b-b436-ae644dfb8acd
137: [INFO] DDSTopologyFile:     /home/dklein/projects/FairMQ/test/sdk/test_topo.xml
137: [INFO] requiredNofAgents:   12
137: [INFO] DDSRMSPlugin:        localhost
137: [INFO] DDSSession running?: 1
137: [       OK ] Topology.Basic (91 ms)
137: [----------] 1 test from Topology (91 ms total)
137:
137: [----------] Global test environment tear-down
137: [==========] 1 test from 1 test case ran. (91 ms total)
137: [  PASSED  ] 1 test.
1/1 Test #137: SDK.Topology.Basic ...............   Passed    0.11 sec

The following tests passed:
        SDK.Topology.Basic

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.12 sec
```

Do you have an idea what is missing?

Reply:
@dennisklein The code looks correct. One thing I would check first is that DDS commander is running. Seems that "mDDSSessionID = mDDSSession.create();" call failed.

Reply:
Thx. With FairRootGroup/DDS@21e8ad68ae195d49a6b8a7ecf5a53b15d6aabf6f and reinstalling after recompiling the `wn_bin` target on my end, it works :)